agent/engine/task_manager.go

```diff
@@ -510,10 +510,10 @@ func (mtask *managedTask) cleanupTask(taskStoppedDuration time.Duration) {
 	// discard events while the task is being removed from engine state
 	go mtask.discardEventsUntil(handleCleanupDone)
 	mtask.engine.sweepTask(mtask.Task)
-	mtask.engine.state.RemoveTask(mtask.Task)
-	log.Debug("Finished removing task data; removing from state no longer managing", "task", mtask.Task)
 	// Now remove ourselves from the global state and cleanup channels
 	mtask.engine.processTasks.Lock()
+	mtask.engine.state.RemoveTask(mtask.Task)
+	seelog.Debug("Finished removing task data; removing from state no longer managing", "task", mtask.Task)
```

Comment:
We need to update the format string when calling seelog directly. I think we should also update the content of the string itself; it seems a bit confusing to me. Maybe something like:
`seelog.Debugf("Finished removing task data, no longer managing task %v", mtask.Task)`